### DISCUSSION TITLE
vAmigaWeb Paula Volume Slider

### DISCUSSION THREAD

**Discussion by aZtOcKdOg** [dirkwhoffmann/vAmiga#787] [Ideas]
Hi, I have been doing a bit of streaming across Amiga emulation distros and the 'Paula Volume Control' of WinUAE / Amiberry has come in extremely handy for levelling audio levels between digital inputs on the stream box!

This volume would be in the gain stage BEFORE the iOS / iPadOS audio volume...

I guess if that hasn't been implemented by the vAmiga core than it would be difficult to jack into but if it is there somewhere then perhaps just a gui slider or numerical indicator dialog.

Just quick thoughts and great updates btw! Lets chat again

**Comment by dirkwhoffmann** [comment]
This issues is related to vAmigaWeb, isn't it? If yes, it's been opened in the wrong repo.

Anyway, here is what the core provides. It allows the user to control the volume of each Paula channel individually as well as to control the "master volume" for the left and right stereo output. The master volume level is applied after the four Paula channels have been mixed together:

<img width="641" alt="Bildschirm­foto 2023-03-24 um 14 39 58" src="https://user-images.githubusercontent.com/12561945/227536581-fa12076e-60d1-41e1-9ba7-6cdba0bd3d46.png">

**Reply by aZtOcKdOg** [reply]
Thanks! Yes oops sorry wrong repo (I was flicking between them to find info) Thanks so much for the info I will pass it on... great work btw

**Comment by aZtOcKdOg** [comment]
Hello o/ A quick follow up re Paula implementation / sampler... is there any plans to add an emulated parallel port 'sampler' input piped through from the host OS audio device I/O... or is there a diff way to obtain the audio input streams into the virtual Amiga? 

Hope not to OT- thanks all and g'day from AU O/

**Comment by aZtOcKdOg** [comment]
I was looking at this:

![image](https://github.com/dirkwhoffmann/vAmiga/assets/64781882/9231fe9f-d969-46c3-88f5-8e5b744c288a)

I did womder if implemented as I will see if mithr. may be able get that feature in vAmigaWeb!

Thanks for any info :)

**Comment by dirkwhoffmann** [comment]
No, vAmiga doesn't emulate the parallel port at all at the moment. The `Sampler` class is responsible to create the output audio stream from the sample stream generated by Paula. Since I am busy with other stuff at the moment, a parallel port sampler is unlikely to be implemented by myself in the near future. Maybe in a longer run...
